tests/syft/core/node/common/plan/plan_test.py

```diff
@@ -203,7 +209,118 @@ def test_plan_batched_execution() -> None:
     ]
 
     for x, y in zip(x_batches, y_batches):
-        plan_pointer(x, y)
+        plan_pointer(x=x, y=y)
 
         # checks if (model(x) == y) == [True, True]
         assert all(result_tensor_pointer3.get())
+
+
+def test_make_plan() -> None:
+    alice = sy.VirtualMachine(name="alice")
+    alice_client = alice.get_root_client()
+
+    @make_plan
+    def add_plan(inp=th.zeros((3))) -> th.Tensor:  # type: ignore
+        return inp + inp
+
+    input_tensor = th.tensor([1, 2, 3])
+    plan_pointer = add_plan.send(alice_client)
+    res = plan_pointer(inp=input_tensor)
+    assert th.equal(res.get()[0], th.tensor([2, 4, 6]))
+
+
+def test_make_plan2() -> None:
+    alice = sy.VirtualMachine(name="alice")
+    alice_client = alice.get_root_client()
+
+    @make_plan
+    def mul_plan(inp=th.zeros((3)), inp2=th.zeros((3))) -> th.Tensor:  # type: ignore
+        return inp * inp2
+
+    t1, t2 = th.tensor([1, 2, 3]), th.tensor([1, 2, 3])
+    plan_pointer = mul_plan.send(alice_client)
+    res = plan_pointer(inp=t1, inp2=t2)
```

Comment:
can this plan be run multiple times? let's add a test to be sure.

Reply:
Yes it can, there is already a nb in this pr doing that. Will add a test